### PR TITLE
[Windows] change I/O check to be PDH based rather than WMI based.

### PR DESCRIPTION
**Pull request by derekwbrown**
WMI based causes high cpu load from the wmi service.

### Motivation

Customer feedback on CPU consumption of the wmi service

### CODE REVIEW

File: pkg/util/winutil/pdhutil/pdhcounter.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+		}
+		winerror = win.PdhAddCounter(p.query, path, uintptr(0), &p.singleCounter)
+		if win.ERROR_SUCCESS != winerror {
+			return nil, fmt.Errorf("Failed to add single counter %d", winerror)
+		}
+	}
+	// do the initial collect now
+	win.PdhCollectQueryData(p.query)
+	return &p, nil
+
```
Comment (gmmeyer) [inline pkg/util/winutil/pdhutil/pdhcounter.go]: sorry about this but can you align the return with the  bottom `}`?

I won't block on this, just a nit

File: pkg/util/winutil/pdhutil/pdhhelper.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+	procPdhEnumObjectItems          = modPdhDll.NewProc("PdhEnumObjectItemsW")
+	procPdhMakeCounterPath          = modPdhDll.NewProc("PdhMakeCounterPathW")
+	procPdhGetFormattedCounterValue = modPdhDll.NewProc("PdhGetFormattedCounterValue")
+)
+
+const (
+	// taken from winperf.h
+	PERF_DETAIL_NOVICE   = 100 // The uninformed can understand it
+	PERF_DETAIL_ADVANCED = 200 // For the advanced user
+	PERF_DETAIL_EXPERT   = 300 // For the expert user
```
Comment (gmmeyer) [inline pkg/util/winutil/pdhutil/pdhhelper.go]: Are these unused? Can they be removed? if there's a good reason for keeping them here it's okay.

File: pkg/util/winutil/winstrings.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+package winutil
+
+import (
+	"bytes"
+)
+
+// ConvertWindowsStringList Converts a windows-style C list of strings
+// (single null terminated elements
+// double-null indicates the end of the list) to an array of Go strings
+func ConvertWindowsStringList(winput []uint16) []string {
```
Comment (gmmeyer) [inline pkg/util/winutil/winstrings.go]: I like the name winput

File: pkg/collector/corechecks/system/iostats_pdh_windows.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+	modkernel32 = syscall.NewLazyDLL("kernel32.dll")
+
+	procGetLogicalDriveStringsW = modkernel32.NewProc("GetLogicalDriveStringsW")
+	procGetDriveType            = modkernel32.NewProc("GetDriveTypeW")
+)
+
+const (
+	ERROR_SUCCESS        = 0
+	ERROR_FILE_NOT_FOUND = 2
+	DRIVE_REMOVABLE      = 2
```
Comment (gmmeyer) [inline pkg/collector/corechecks/system/iostats_pdh_windows.go]: `DRIVE_REMOVABLE` and `ERROR_FILE_NOT_FOUND` don't seem to be used. should we keep them or can we remove them? It's cool if we should keep them, just curious

**Review by gmmeyer — DISMISSED** [review]
this looks good to me, a few nits, but feel free to merge. they're unimportant

File: pkg/collector/corechecks/system/iostats_pdh_windows.go (hunk 4 above)
Comment (derekwbrown) [reply pkg/collector/corechecks/system/iostats_pdh_windows.go]: Those are the other constants for GetDriveType(), so they're there for completeness.

File: pkg/util/winutil/pdhutil/pdhhelper.go (hunk 2 above)
Comment (derekwbrown) [reply pkg/util/winutil/pdhutil/pdhhelper.go]: Again, more for completeness than anything else.  We're providing access to the full API, so I thought we should have all the flags.

**Pull request merged**